FrameworkSystem/private/standardLogging/gLogging.py

```diff
@@ -0,0 +1,335 @@
+    This option will not be modified anymore. 
+    The options of the childrens will be updated if they were not modified before by a developer
+    :params optionName: string representing the name of the option to modify
+    :params value: boolean to give to the option  
+    """
+    self._options[optionName] = value
+    self._optionsmodified[optionName] = True
+    self._setChildrensDisplayOptions(optionName, self._options)
+    self._updateFormat()
+
+  def registerBackends(self, desiredBackends, backendOptions=None):
+    """
+    Attach a list of backends to the gLogging object.
+    :params desiredBackends: a list of different names attaching to differents backends.
+                             example : ['stdout', 'file', 'server']
+    :params backendOptions: a dictionary of different backend options. 
+                            example: {'FileName': '/tmp/log.txt'}
+    """
+    for stringHandler in desiredBackends:
+      stringHandler = stringHandler.lower()
+      stringHandler = stringHandler.strip()
+
+      # check if the name is correct
+      if stringHandler in self._backendsDict:
+        backend = self._backendsDict[stringHandler]
+
+        # check if the backend is already in the list
+        if backend not in self._backendsList:
+          if backendOptions is not None:
+            backend.setParameters(backendOptions.copy())
+
+          backend.configureHandler()
+          self._logger.addHandler(backend.getHandler())
+          self._backendsList.append(backend)
+          self._updateFormat()
+      else:
+        self._updateFormat()
+        self.warn("%s is not a valid backend name.", stringHandler)
+
+  def setLevel(self, levelName):
+    """
+    Set a level to the logger.
+    :params levelName: string representing the level to give to the logger
+    :return: boolean representing if the setting is done or not
+    """
+    result = False
+    levelName = levelName.upper()
+    if levelName in LogLevels.getLevelNames():
+      self._logger.setLevel(LogLevels.getLevelValue(levelName))
+      result = True
+    return result
+
+  def getLevel(self):
+    """
+    :return: the name of the level
+    """
+    return LogLevels.getLevel(self._logger.getEffectiveLevel())
+
+  def shown(self, levelName):
+    """
+    Determine if messages with a certain level will be displayed or not.
+    :params levelName: string representing the level to analyse
+    :return: boolean which give the answer
+    """
+    result = False
+    levelName = levelName.upper()
+    if levelName in LogLevels.getLevelNames():
+      result = self._logger.isEnabledFor(LogLevels.getLevelValue(levelName))
+    return result
+
+  @staticmethod
+  def getName():
+    """
+    :return: "system name/component name"
+    """
+    return gLogging._componentName
+
+  def getSubName(self):
+    """
+    :return: the name of the logger
+    """
+    name = self._logger.name
+    name = name.replace("root", "")
```

Comment:
I can't because the name qualifies the logger, so if I change the name, I change the logger. I tried to change it before, it was inconclusive. 
But I created a name attribute that stores the subName of the logger. It is more simple now